pkg/controller/sub_controller/cn/pod.go

```diff
@@ -0,0 +1,39 @@
+package cn
+
+import (
+	"context"
+	v1 "github.com/selectdb/doris-operator/api/v1"
+	"github.com/selectdb/doris-operator/pkg/common/utils/resource"
+	corev1 "k8s.io/api/core/v1"
+	"strconv"
+)
+
+func (cn *Controller) buildCnPodTemplateSpec(dcr *v1.DorisCluster) corev1.PodTemplateSpec {
+	podTemplateSpc := resource.NewPodTemplateSpc(dcr, v1.Component_CN)
+	var containers []corev1.Container
+	containers = append(containers, podTemplateSpc.Spec.Containers...)
+	cnContainer := cn.cnContainer(dcr)
+	containers = append(containers, cnContainer)
+	podTemplateSpc.Spec.Containers = containers
+	return podTemplateSpc
+}
+func (cn *Controller) cnContainer(dcr *v1.DorisCluster) corev1.Container {
+	container := resource.NewBaseMainContainer(dcr.Spec.CnSpec.BaseSpec, v1.Component_CN)
+	cnConfig, _ := cn.GetConfig(context.Background(), &dcr.Spec.CnSpec.ConfigMapInfo, dcr.Namespace)
+	address := v1.GetConfigFEAddrForAccess(dcr, v1.Component_CN)
+	var feconfig map[string]interface{}
+	if address == "" {
+		if dcr.Spec.CnSpec.ConfigMapInfo.ConfigMapName != "" && dcr.Spec.CnSpec.ConfigMapInfo.ResolveKey != "" {
+			feconfig, _ = cn.getFeConfig(context.Background(), &dcr.Spec.CnSpec.ConfigMapInfo, dcr.Namespace)
+		}
```

Comment:
fe query port set in cn config does not occur in the address is empty. fe query port set has nothing to do with the address. directly use getFeConfig for query port get.